domain/Lockset.go

```diff
@@ -2,40 +2,33 @@ package domain
 
 import (
 	"go/token"
+
 	"golang.org/x/tools/go/ssa"
 )
+
 type locksLasUse map[token.Pos]*ssa.CallCommon
 
 type Lockset struct {
 	ExistingLocks   locksLasUse
 	ExistingUnlocks locksLasUse
 }
 
-
 func NewLockset() *Lockset {
 	return &Lockset{
-		ExistingLocks:   make(locksLasUse, 0),
-		ExistingUnlocks: make(locksLasUse, 0),
+		ExistingLocks:   make(locksLasUse),
+		ExistingUnlocks: make(locksLasUse),
 	}
 }
 
 func (ls *Lockset) UpdateLockSet(newLocks, newUnlocks locksLasUse) {
-	if newLocks != nil {
-		for lockName, lock := range newLocks {
-			ls.ExistingLocks[lockName] = lock
-		}
-	}
 	for unlockName, _ := range newUnlocks {
-		if _, ok := ls.ExistingLocks[unlockName]; ok {
-			delete(ls.ExistingLocks, unlockName)
-		}
+		delete(ls.ExistingLocks, unlockName)
 	}
 
-	if newUnlocks != nil {
-		for unlockName, unlock := range newUnlocks {
-			ls.ExistingUnlocks[unlockName] = unlock
-		}
+	for unlockName, unlock := range newUnlocks {
+		ls.ExistingUnlocks[unlockName] = unlock
 	}
+
 	for lockName, _ := range newLocks {
```

Comment:
Also, if possible, can you change this block as well and trim all the redundant code from it as you did to the other blocks?

Reply:
✅